hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -8,14 +8,285 @@ title: "OTP Controller Technical Specification"
+
+In order to enable simulation and FPGA emulation of the OTP controller even without access to the proprietary OTP IP, a generalized and synthesizable model of the OTP IP is provided in the form of a generic technology primitive (`prim_generic_otp`).
+
 
 # Theory of Operations
 
-## Block Diagram
+Conceptually speaking, the OTP functionality is at a high level split into "front-end" and "back-end".
+The "front-end" contains the logical partitions and associated buffer registers that feed the hardware and software consumer interfaces of the system.
+The "back-end" represents the programming interface used by hardware and software components to stage the upcoming values.
+The diagram below illustrates this behavioral model.
+
+![OTP Controller Block Diagram](otp_behavioral_model.svg)
+
+Note that the front-end contains both buffered and unbuffered partitions.
+Buffered partitions are sensed once per power cycle and their contents are stored in registers, whereas unbuffered partitions are read on-demand.
+The former are typically partitions that contain data like hardware configuration bits, key material and the life cycle state that need to be always available to the hardware, whereas the latter are large partitions that are accessed infrequently, such as the software configurations.
+Values that are programmed into a buffered partition via the programming interface (coupled with read verification) are merely "staged", and do not take effect until the next power cycle.
+
+The sections below describe the operation of various pieces of the OTP controller and how it supports the described functionality.
+
+## Logical Partitions
+
+The OTP is logically separated into partitions that represent different functions.
+This means the isolation is virtual and maintained by the OTP controller instead of the underlying OTP IP.
+
+Within each logical partition, there are specific enforceable properties
+
+- Secrecy
+  - This controls whether a particular partition contains secret data
+  - If secret, a partition is not readable by software once locked, and is encrypted in storage
+- Read lockability
+  - This controls whether a particular partition disables software readability for later stage software.
+  - Some partitions can be locked statically (by computing and storing an associated digest in OTP), others can be read locked at runtime via a CSRs.
+- Write lockability
+  - This controls whether a partition is locked and prevented from future updates.
+  - A locked partition is stored alongside a 32b digest (truncated hash) to be used later for integrity verification.
+- Integrity Verification
+  - Once a partition is locked, it can undergo periodic verification (time-scale configurable by software)
```

Comment:
When you mention once a partition is locked -> does this `locked` applies to both read lock and write lock, or it is write lock only?

Reply:
Integrity verification applies to partitions that get write-locked after burning in a digest value.

Note that for read locks there are two mechanisms as described above. SW_CFG can be locked at runtime via a CSR. secret partitions get statically read locked whenever they get write locked (these read and write locks are coupled to the same digest.

Reply:
I'll amend that sentence to make it clear.